src/qat/purr/compiler/validation_passes.py

```diff
@@ -0,0 +1,145 @@
+from numbers import Number
+from typing import List
+
+import numpy as np
+
+from qat import qatconfig
+from qat.ir.pass_base import ValidationPass
+from qat.ir.result_base import ResultManager
+from qat.purr.compiler.builders import InstructionBuilder
+from qat.purr.compiler.devices import MaxPulseLength, PulseChannel, Qubit
+from qat.purr.compiler.execution import QuantumExecutionEngine
+from qat.purr.compiler.hardware_models import QuantumHardwareModel
+from qat.purr.compiler.instructions import (
+    Acquire,
+    AcquireMode,
+    CustomPulse,
+    PostProcessing,
+    ProcessAxis,
+    Pulse,
+    Sweep,
+    Variable,
+)
+
+
+class InstructionValidation(ValidationPass):
+    """
+    Extracted from QuantumExecutionEngine.validate()
+    """
```

Comment:
In cases like this, could also have a minor description explaining what the pass does/checks? :smile:

Reply:
Good point. It was taken as is from legacy code on the engine, I think it's bulky and definitely needs dilution. Will see if I have time to break it up into meaningful pieces. Thanks !

Reply:
Decided to leave it as is for this PR. Changes will come in future polishing sweeps